packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,2300 @@
+              DiagnosticsProperty<WebImageProvider>('Image provider', this),
+              DiagnosticsProperty<String>('Image src', key, defaultValue: null),
+            ];
+            return true;
+          }());
+          FlutterError.reportError(FlutterErrorDetails(
+            context: ErrorDescription('while checking the cache location of an image'),
+            informationCollector: collector,
+            exception: exception,
+            stack: stack,
+          ));
+          completer.complete();
+        }
+      },
+    );
+    return completer.future;
+  }
+
+  /// This method is used by both [resolve] and [obtainCacheStatus] to ensure
+  /// that errors thrown during key creation are handled whether synchronous or
+  /// asynchronous.
+  void _createErrorHandler(
+    _KeyAndErrorHandlerCallback successCallback,
+    _AsyncKeyErrorHandler errorCallback,
+  ) {
+    bool didError = false;
+    Future<void> handleError(Object exception, StackTrace? stack) async {
+      if (didError) {
+        return;
+      }
+      if (!didError) {
+        didError = true;
+        errorCallback(src, exception, stack);
+      }
+    }
+
+    try {
+      successCallback(src, handleError);
+    } catch (error, stackTrace) {
+      handleError(error, stackTrace);
+    }
+  }
+
+  /// Called by [resolve] with the key as [src].
+  ///
+  /// The implementation uses the key to interact with the [WebImageCache],
+  /// calling [WebImageCache.putIfAbsent] and notifying listeners of the
+  /// [stream].
+  @protected
+  void resolveStreamForKey(
+      WebImageStream stream, String key, ImageErrorListener handleError) {
+    // This is an unusual edge case where someone has told us that they found
+    // the image we want before getting to this method. We should avoid calling
+    // load again, but still update the image cache with LRU information.
+    if (stream.completer != null) {
+      final WebImageStreamCompleter? completer = webImageCache.putIfAbsent(
+        key,
+        () => stream.completer!,
+        onError: handleError,
+      );
+      assert(identical(completer, stream.completer));
+      return;
+    }
+    final WebImageStreamCompleter? completer = webImageCache.putIfAbsent(
+      key,
+      () => loadImage(key),
+      onError: handleError,
+    );
+    if (completer != null) {
+      stream.setCompleter(completer);
+    }
+  }
+
+  /// Creates a completer which completes if either (1) the image bytes can be
+  /// fetched normally or (2) an <img> tag has decoded the image.
+  WebImageStreamCompleter loadImage(String key) {
+    return OneFrameWebImageStreamCompleter(
+      checkIfImageBytesCanBeFetched(key),
+      _decodeImageTag(),
+    );
+  }
+
+  Future<WebImageInfo> _decodeImageTag() async {
+    final web.HTMLImageElement img =
+        web.document.createElement('img') as web.HTMLImageElement;
+    img.src = src;
+    img.style
+      ..width = '100%'
+      ..height = '100%';
```

Comment:
@harryterkelsen 
My suggestion has been ignored so far, but please think about it again.
I'm seriously sure that the possibility to pass CSS filter is useful and important for multiple cases, to match CORS <img> fallback case with a normal Image widget. So it should be:
```dart
if (cssFilter != null && cssFilter.isNotEmpty) {
    img.style.filter = cssFilter;
  }
```

Reply:
We should support styling with a CSS filter the same way we support opacity (or transforms). Rather than setting an opacity directly on the HTML element, you would wrap the HTMLElementView in an Opacity widget and let the engine handle the opacity for you. Similarly for filters, you should let the engine handle the filter for you by wrapping the widget in (e.g.) an ImageFiltered widget.

Reply:
@harryterkelsen 
Not completely following what exactly you are proposing.
For example, for Image widget there are the following properties: color, opacity, colorBlendMode
So for WebImage they won't be provided and I always have to use ImageFiltered over WebImage to change color? And it will work the same way for WebImage "underlying" Image or HTMLImageElement ?

Reply:
Yes, I agree. We should support `color`, `opacity`, and `colorBlendMode` on `WebImage`. But I don't want to expose `cssFilter` directly since it will be part of the API forever. Adding `color`, `colorBlendMode` and `opacity` can be done later, this PR is intended to be a minimum viable product for developers who want a drop-in replacement for simple usages of `Image.network` that don't work on the web because of CORS restrictions.